### ISSUE TITLE
Why the `detach` in the 

### ISSUE THREAD

**Issue by yohan-pg** [clovaai/stargan-v2#99]
In the code for the diversity sensitive loss:

```
# diversity sensitive loss
if z_trgs is not None:
    s_trg2 = nets.mapping_network(z_trg2, y_trg)
else:
    s_trg2 = nets.style_encoder(x_ref2, y_trg)
x_fake2 = nets.generator(x_real, s_trg2, masks=masks)
x_fake2 = x_fake2.detach()
loss_ds = torch.mean(torch.abs(x_fake - x_fake2))
```

I'm wonder why x_fake2 detached is  here? This is not mentioned in the paper and I really don't see any reason why leaving it attached would pose a problem.
